column-count: 1 should create a multi-column container

fantasai opened the pull request:
#### e10b646accd306950141c89cbd4ef2697e44d6f8
<pre>
column-count: 1 should create a multi-column container
<a href="https://bugs.webkit.org/show_bug.cgi?id=299836">https://bugs.webkit.org/show_bug.cgi?id=299836</a>
<a href="https://rdar.apple.com/161611444">rdar://161611444</a>

Reviewed by Sam Weinig and Antti Koivisto.

Per spec, we need to create a multi-column container whenever either
column-count or column-width is not &apos;auto&apos;. That includes column-count: 1;
  <a href="https://www.w3.org/TR/css-multicol-1/#the-multi-column-model">https://www.w3.org/TR/css-multicol-1/#the-multi-column-model</a>

* LayoutTests/TestExpectations:

Adjust test failures due to actually doing multicol layout for 1 column.
New failures:
- monolithic-overflow-*.tentative failures due to test errors
  <a href="https://github.com/web-platform-tests/wpt/issues/55152">https://github.com/web-platform-tests/wpt/issues/55152</a>
- multicol-span-all failures due to painting errors
  <a href="https://bugs.webkit.org/show_bug.cgi?id=299854">https://bugs.webkit.org/show_bug.cgi?id=299854</a>
- overflow-clip-010 due to some weird &apos;overflow: clip&apos; bug
  <a href="https://bugs.webkit.org/show_bug.cgi?id=299857">https://bugs.webkit.org/show_bug.cgi?id=299857</a>
- resize-in-strict-containment-nested because we can&apos;t fragment multicol well
  <a href="https://bugs.webkit.org/show_bug.cgi?id=269202">https://bugs.webkit.org/show_bug.cgi?id=269202</a>

* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::willCreateColumns const):

Canonical link: <a href="https://commits.webkit.org/300787@main">https://commits.webkit.org/300787@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/11f6c11815346eb0ad125c485452845b1b75eff0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123859 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43574 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34271 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130645 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76010 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/dae8cb82-b88e-4dda-9381-94942dd43760) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125736 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44298 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52168 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94200 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62507 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d5ded122-fdd5-4cee-8377-9a38ff7f7c97) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126812 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35288 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110794 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74798 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7820d877-0290-46ba-a528-eb13ab69edbd) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34244 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28955 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74122 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105015 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29178 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133339 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50812 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38692 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102670 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51186 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107014 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102498 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26061 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47836 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26092 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/47661 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50665 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56429 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50139 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53485 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51813 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/93c6152a8c30967102777b9e926caf2717f71290. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51536

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6e9f888751e2c7120db6b4bcef9148a63cf141f8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51536

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/11f6c11815346eb0ad125c485452845b1b75eff0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51536

fantasai committed:
column-count: 1 should create a multi-column container
https://bugs.webkit.org/show_bug.cgi?id=299836
rdar://161611444

Reviewed by Sam Weinig and Antti Koivisto.

Per spec, we need to create a multi-column container whenever either
column-count or column-width is not 'auto'. That includes column-count: 1;
  https://www.w3.org/TR/css-multicol-1/#the-multi-column-model

* LayoutTests/TestExpectations:

Adjust test failures due to actually doing multicol layout for 1 column.
New failures:
- monolithic-overflow-*.tentative failures due to test errors
  https://github.com/web-platform-tests/wpt/issues/55152
- multicol-span-all failures due to painting errors
  https://bugs.webkit.org/show_bug.cgi?id=299854
- overflow-clip-010 due to some weird 'overflow: clip' bug
  https://bugs.webkit.org/show_bug.cgi?id=299857
- resize-in-strict-containment-nested because we can't fragment multicol well
  https://bugs.webkit.org/show_bug.cgi?id=269202

* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::willCreateColumns const):

Canonical link: https://commits.webkit.org/300787@main
